core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,290 @@
+      .and.returnValue(Promise.resolve(userInfoCanCreateCollection));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    spyOn(component, 'initCreationProcess');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(component.initCreationProcess).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(true);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should initialize and create new exploration if user cannot' +
+    ' create collections', fakeAsync(() => {
+    const userServiceSpy = spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfoCanNotCreateCollection));
+    const urlParamsSpy = spyOn(urlService, 'getUrlParams').and.returnValue({
+      mode: 'create'
+    });
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(userServiceSpy).toHaveBeenCalled();
+    expect(urlParamsSpy).toHaveBeenCalled();
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+    expect(component.canCreateCollections).toBe(false);
+    expect(component.userIsLoggedIn).toBe(true);
+  }));
+
+  it('should stop the creation process if another' +
+    ' creation is in progress', () => {
+    component.creationInProgress = true;
+    expect(component.initCreationProcess()).toBe();
+  });
+
+  it('should create new exploration if user cannot create collections', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = false;
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+  });
+
+  it('should redirect user to create new exploration when user clicks' +
+    ' create button and is not on creator dashboard page', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = true;
+    const urlServiceSpy = spyOn(urlService, 'getPathname').and.returnValue(
+      'not/creator-dashboard');
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      location: {
+        replace: (val: string) => {}
+      }
+    });
+    const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
+
+    component.initCreationProcess();
+
+    expect(urlServiceSpy).toHaveBeenCalled();
+    expect(replaceSpy).toHaveBeenCalledWith('/creator-dashboard?mode=create');
+  });
+
+  it('should open a create activity modal if user' +
```

Comment:
You should have parity in the checks for similar tests. E.g. for this one there should be a check that replaceSpy is not called.

Reply:
Sorry -- was this description changed from before? Now this test isn't testing the opposite of the previous test (based on the description) so I would withdraw the replaceSpy comment. Though, in that spirit, it might make sense to have a separate test along the lines of: "it should not redirect user when user clicks create button and is on creator dashboard page".

Reply:
Those might essentially be the same tests, only differing by the replaceSpy line. I have changed the description to `should not redirect user but open a create activity modal if user can create collections and is on creator dashboard page` to accommodate both tests.